### PR TITLE
[Fix] Small bugs

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description

This PR fixes:
- Homepage wording
- Handles error messages for contract call and lambda parsing
- Show error when mutez amounts are `<= 0`
- Show the exact proposal duration time
- Show an error when importing a non TzSafe contract
- Detecting correctly FA2 transactions

### CODE REVIEW

File: components/transferForm.tsx [hunk 1]
```diff
@@ -493,7 +493,10 @@ function Basic({
         if (!exists) {
           errors.walletAddress = `Contract does not exist at address ${values.walletAddress}`;
         }
-        if (isNaN(Number(values.amount))) {
+
+        const convertedValue = Number(values.amount);
+
+        if (isNaN(convertedValue) || convertedValue <= 0) {
```
Comment (rueshyna) [inline components/transferForm.tsx:499]: we should also check the amount in "Fund wallet"

File: components/transferForm.tsx [hunk 2]
```diff
@@ -514,7 +517,7 @@ function Basic({
                 <Field
                   name="amount"
                   className=" w-full rounded p-2 text-black"
-                  placeholder="0"
```
Comment (rueshyna) [inline components/transferForm.tsx:517]: the amount in "Contract Execution" is an exception that can be `0` but not `<0`

File: components/import/basic.tsx [hunk 3]
```diff
@@ -69,32 +74,44 @@ function Basic() {
...
-          ...values,
-          validators,
-          requiredSignatures: storage.threshold.toNumber(),
-          effectivePeriod: storage.effective_period.toNumber(),
-        };
-        setFormState(data as any);
-        setActiveStepIndex(activeStepIndex + 1);
-        setIsLoading(false);
+          if (version === "unknown version") {
+            throw new Error("The contract is not a TzSafe contract");
```
Comment (rueshyna) [inline components/import/basic.tsx:88]: this error wasn't handled on UI.
import `KT1BMLhaeartDoLXjiQZorD3yKQj6SoH6c5R` to reproduce

File: components/import/basic.tsx [hunk 4]
```diff
@@ -69,32 +74,44 @@ function Basic() {
...
+          }
+
+          let balance = await state?.connection.tz.getBalance(
+            values.walletAddress
+          );
+          let v = toStorage(version, storage, balance);
```
Comment (rueshyna) [inline components/import/basic.tsx:94]: is it possible to have errors here? if yes, we should handle it.

File: versioned/interface.ts [hunk 5]
```diff
@@ -161,9 +161,9 @@ abstract class Versioned {
               }
               const p = new Parser();
               try {
-                const _ = p.parseScript(x);
+                p.parseScript(x);
               } catch {
-                return "Bad lambda";
+                return "Unable to parse the lambda";
```
Comment (rueshyna) [inline versioned/interface.ts:166]: In which case, I'll see this error on UI? It doesn't seem to work on UI.

**Review by rueshyna — COMMENTED** [review]
although the placeholder of amount is 1, but the default value is still 0.

File: components/import/basic.tsx (hunk 3 above)
Comment (JulesGuesnon) [reply components/import/basic.tsx:88]: Hum are you sure it's the right contract? I imported it and I got the error saying: `The contract is not a TzSafe contract`

File: components/import/basic.tsx (hunk 4 above)
Comment (JulesGuesnon) [reply components/import/basic.tsx:94]: Everything is in a try/catch so it's already handled

File: versioned/interface.ts (hunk 5 above)
Comment (JulesGuesnon) [reply versioned/interface.ts:166]: You can try with this input `qdfqdf!`, it's working on my side. Also you may need to blur the input to trigger a check

File: components/import/basic.tsx (hunk 3 above)
Comment (rueshyna) [reply components/import/basic.tsx:88]: It shows up on UI now :thinking: (It was nothing there)

File: versioned/interface.ts (hunk 5 above)
Comment (rueshyna) [reply versioned/interface.ts:166]: close the thread, as the new request is to remove lambda execution.

**Review by rueshyna — APPROVED** [review]

**Pull request merged**